CoreDistributedCache/NHibernate.Caches.CoreDistributedCache/CoreDistributedCache.cs

```diff
@@ -25,16 +25,115 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Extensions.Caching.Distributed;
 using NHibernate.Caches.Util;
 using NHibernate.Util;
 
 namespace NHibernate.Caches.CoreDistributedCache
 {
+	// 6.0 TODO: replace that class by its base
 	/// <summary>
 	/// Pluggable cache implementation using <see cref="IDistributedCache"/> implementations.
 	/// </summary>
-	public partial class CoreDistributedCache : ICache
+	public class CoreDistributedCache : CoreDistributedCacheBase,
```

Comment:
For avoiding binary breaking changes, I only see this ugly solution.

Removing a non-virtual method for a virtual inherited one is a breaking change, as some caller may try to `call` it, as explained [here](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#members):

> Adding virtual to a member
> While this change would often work without breaking too many scenarios because C# compiler tends to emit `callvirt` IL instructions to call non-virtual methods (`callvirt` performs a null check, while a normal `call` won't), we can't rely on it. C# is not the only language we target and the C# compiler increasingly tries to optimize `callvirt` to a normal `call` whenever the target method is non-virtual and the `this` is provably not null (such as a method accessed through the `?.` null propagation operator). Making a method virtual would mean that consumer code would often end up calling it non-virtually.

And we are in a worst case: we have removed many of these members (all the async ones for all the other caches), counting on virtual inherited ones for replacing them.

Moreover the old `void Lock` cannot coexist with `object Lock` in the same class, so we anyway need an "intermediate" class.